ui: FloatingTaskTimeEstimateModal

Emilia-Z opened the pull request.

sol-wizard commented:
LGTM

Emilia-Z committed:
ui: FloatingTaskTimeEstimateModal

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-card.tsx:
```diff
@@ -22,16 +25,82 @@ export const FloatingTaskCard = ({
   };
 
+  const formatDuration = (duration: string | number | undefined): string | null => {
+  if (duration == null) return null;
```
try to simplify this. it looks too complicated. there are unnecessary validation. like duration type can't be number. also extract it into a helper function and put in util folder. i think we already have one function called convertSubtaskTimeForm. see whether you can user that

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-card.tsx:
```diff
@@ -22,16 +25,82 @@ export const FloatingTaskCard = ({
+  const handleEstimateTime = async (task: FloatingTaskDTO) => {
+    try {
+      setError(null);
+      setEstimate(null);
```
if you already use setError(null);
    setEstimate(null); at closeModal, why do you need it here again

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-card.tsx:
```diff
@@ -22,16 +25,82 @@ export const FloatingTaskCard = ({
+      };
+
+      const result = await estimateTaskTime(payload);
+      console.log("Estimation result:", result);
```
remove console log

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-time-estimate-modal.tsx:
```diff
@@ -0,0 +1,102 @@
+  durationText?: string;
+  error?: string | null;
+  onPickTime?: () => void;
+  onAddToSlot?: () => void;
```
looks like you don't need onPickTime?: () => void;
Comment view  onAddToSlot?: () => void;

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-time-estimate-modal.tsx:
```diff
@@ -0,0 +1,102 @@
+  onAddToSlot,
+  isEstimating,
+}: Props) => {
+  const hasError = !!error;
```
why need a separate variable here

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-time-estimate-modal.tsx:
```diff
@@ -0,0 +1,102 @@
+      <Pressable
+        className="flex-1 bg-black/40 items-center justify-center px-6"
+        onPress={onClose}
+      >
```
i think there is a way to do a backdrop for modal rather than use a big pressable, search how to do that

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-time-estimate-modal.tsx:
```diff
@@ -0,0 +1,102 @@
+          {!isEstimating && !hasError && durationText && (
+            <>
+              <Text className="text-xl leading-6 text-gray-800 font-baloo">
+                We&apos;ve estimated this task will take around{" "}
```
delete{" "}

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-time-estimate-modal.tsx:
```diff
@@ -0,0 +1,102 @@
+            <>
+              <Text className="text-xl leading-6 text-gray-800 font-baloo">
+                We&apos;ve estimated this task will take around{" "}
+                <Text className="text-lime-300 font-baloo">
```
looks like you already have font-baloo in the parent text?

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-time-estimate-modal.tsx:
```diff
@@ -0,0 +1,102 @@
+                >
+                  <Text className="text-xs text-gray-800 font-balooExtraBold">
+                    Add to slot
+                  </Text>
```
<img width="262" height="244" alt="Image" src="https://github.com/user-attachments/assets/b46544eb-1af9-40aa-a721-5743b265e6bc" />

the button looks strange?
also i don't feel like this is the right green color. go to figma and check the color

Emilia-Z committed:
fix(modal): use react-native-modal backdrop and clean up logic

- Replace pressable backdrop with react-native-modal
- Remove redundant state and console logs
- Simplify error handling
- Align modal UI styles with design

Emilia-Z committed:
Merge branch 'main' into floating-task-estimate-time-modal-ui

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-time-estimate-modal.tsx:
```diff
@@ -0,0 +1,88 @@
+      style={{ margin: 0 }}  
+    >
+      <View className="flex-1 items-center justify-center px-6">
+        <View className="w-full max-w-[280px] rounded-3xl bg-background px-[40px] py-[50px]">
```
for  px-[40px] py-[50px], usually we don't use a fixed value for css because this couldn't change according to differenct size of the screen.

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-time-estimate-modal.tsx:
```diff
@@ -0,0 +1,88 @@
+
+              <View className="mt-8 flex-row items-center justify-end">
+                <Pressable onPress={onClose}>
+                  <Text className="text-sm text-[#8C8C8C] font-baloo">
```
I think #8C8C8C has a value in tailwind config. you can use text-primary to replace this

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-card.tsx:
```diff
@@ -22,16 +26,48 @@ export const FloatingTaskCard = ({
+  const [isEstimating, setIsEstimating] = useState(false);
+  const [estimate, setEstimate] = useState<TaskTimeEstimation | null>(null);
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [error, setError] = useState<string | null>(null);
```
it seems to have too many props here. if you want to learn how to use react query, you can use usemutation to wrap await estimateTaskTime, then it will help you handle all the isEstimating and error logic. check blotztask-mobile/src/shared/hooks/useTaskMutations.ts you can see how to expose the state through tan stack query. you can create a separate useEstimateTaskTime and put it in the hook folder.

nxn-nicole commented on blotztask-mobile/src/feature/star-spark/components/floating-task-card.tsx:
```diff
@@ -22,16 +26,48 @@ export const FloatingTaskCard = ({
+
+      const result = await estimateTaskTime(payload);
+
+      const formatted = convertSubtaskTimeForm(result.duration);
```
formatted isn't a good naming

Emilia-Z committed:
refactor(task-estimation): move async logic to react-query and align UI styles

Emilia-Z committed:
Merge branch 'main' into floating-task-estimate-time-modal-ui

Emilia-Z committed:
fix: remove unused packages

Emilia-Z committed:
Merge branch 'floating-task-estimate-time-modal-ui' of https://github.com/sol-wizard/Blotz-Task-App into floating-task-estimate-time-modal-ui

nxn-nicole committed:
Merge branch 'main' into floating-task-estimate-time-modal-ui
